pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -235,7 +235,7 @@ func (r PvcCloneReconciler) prepare(syncRes *dataVolumeSyncResult) error {
 	if err := r.populateSourceIfSourceRef(dv); err != nil {
 		return err
 	}
-	if isCrossNamespaceClone(dv) && dv.Status.Phase == cdiv1.Succeeded {
```

Comment:
why cleanup when not needed?

Reply:
just moved the cross ns check to https://github.com/kubevirt/containerized-data-importer/blob/2137848f19209350b1a28fa8958955839e7f6dcf/pkg/controller/datavolume/pvc-clone-controller.go#L714-L719